lib/logstash/filters/grok.rb

```diff
@@ -264,8 +278,13 @@ def filter(event)
       done = false
 
       @logger.debug? and @logger.debug("Running grok filter", :event => event);
+
+      reasons = Set.new
+      
       @patterns.each do |field, groks|
-        if match(groks, field, event)
+        success, reason = match(groks, field, event)
```

Comment:
I feel like the only reason you have to carry this `reason` attribute is because error handling is prematurely being handled within `def match`. If we bring the exception management here, would it make things more straightforward?